predictsignauxfaibles/__main__.py

```diff
@@ -1,80 +1,207 @@
-from datetime import datetime
+# pylint: disable=invalid-name
 import logging
 import sys
 
-from sklearn.metrics import balanced_accuracy_score
+from os import path
+import argparse
+import importlib.util
 
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from sklearn.metrics import balanced_accuracy_score
 from predictsignauxfaibles.pipelines import run_pipeline
+from predictsignauxfaibles.data import OversampledSFDataset, SFDataset
 
 sys.path.append("../")
-import models.default.model_conf as conf  # pylint: disable=wrong-import-position
 
 logging.basicConfig(format="%(asctime)s - %(levelname)s - %(message)s", level="INFO")
 
 # Mute logs from sklean_pandas
 logging.getLogger("sklearn_pandas").setLevel(logging.WARNING)
 
 
-def run():
+def load_conf(args):  # pylint: disable=redefined-outer-name
+    """
+    Loads a model configuration from a argparse.Namespace
+    containing a model name and a configuration filename
+    Args:
+        conf_args: a argparse.Namespace object containing attributes
+            model_name
+            model_conf_path
+    """
+    conf_filepath = path.join("models", args.model_name, args.model_conf_path + ".py")
+
+    spec = importlib.util.spec_from_file_location(
+        f"models.{args.model_name}.{args.model_conf_path}", conf_filepath
+    )
+    model_conf = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(model_conf)  # pylint: disable=wrong-import-position
+    return model_conf
+
+
+def run(args):  # pylint: disable=redefined-outer-name
     """
     Run model
     """
     logging.info(f"Running Model {conf.MODEL_ID} (commit {conf.MODEL_GIT_SHA})")
 
     step = "[TRAIN]"
+    TRAIN_DATASET = OversampledSFDataset(
```

Comment:
if we create the datasets in the same module they are used, they should be lowercased (`train_dataset`) ; i'm actually surprised that pylint didn't catch this

Reply:
✅